app/controllers/application_controller.rb

```diff
@@ -154,6 +156,6 @@ def pagination_offset
   end
 
   def params_for_update
-    body_params.permit(*api_doc_definition.all_attributes - api_doc_definition.read_only_attributes)
+    body_params.except(*BLACKLIST_PARAMS).permit(*api_doc_definition.all_attributes - api_doc_definition.read_only_attributes)
```

Comment:
I don't think this is needed, without it:

```
curl -X PATCH http://localhost:3000/api/v1.0/sources/1 -d '{"tenant": "5678"}' 2>/dev/null | python -m json.tool
{
    "errors": [
        {
            "detail": "found unpermitted parameter: :tenant",
            "status": 400
        }
    ]
}
```

and
```
curl -X PATCH http://localhost:3000/api/v1.0/sources/1 -d '{"tenant_id": "2"}' 2>/dev/null | python -m json.tool
{
    "errors": [
        {
            "detail": "properties tenant_id are not defined in #/components/schemas/Source",
            "status": 400
        }
    ]
}
```

Reply:
@agrare I see that, but do we want to return the 400 here? Was thinking we'd still strip out the `tenant` param, essentially ignoring it so that it would still succeed. What do you think?

Reply:
Oh, I don't know...if it is read-only then I think trying to update it should be a bad request error.  @bdunne?

Reply:
I agree, I think we need to return a 400 here because we say [additional properties are not allowed](https://github.com/ManageIQ/sources-api/blob/42b6081347b45958898121f39197007b8b1d23ad/public/doc/openapi-3-v1.0.0.json#L1366)

Reply:
Thanks, Adam, Brandon, I'll change it